### ISSUE TITLE
fit(data) exemple throws TypeError: ('Values: Expected type int or type float, got ', dtype('int64'))

### ISSUE THREAD

**Issue by anapinkg** [pgmpy/pgmpy#677]
From the pgmpy API reference, I tried the example under `fit(data)`:

``` python
import numpy as np
import pandas as pd`
from pgmpy.models import BayesianModel
values = pd.DataFrame(np.random.randint(low=0, high=2, size=(1000, 5)), columns=['A', 'B', 'C', 'D', 'E'])
model = BayesianModel([('A', 'B'), ('C', 'B'), ('C', 'D'), ('B', 'E')])
model.fit(values)
model.get_cpds()
```

and I got the error

``` bash
> TypeError: ('Values: Expected type int or type float, got ', dtype('int64'))

**Traceback:**
  File "<ipython-input-11-bb819455e71e>", line 1, in <module>
    runfile('/home/ana/.spyder2-py3/temp.py', wdir='/home/ana/.spyder2-py3')

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 699, in runfile
    execfile(filename, namespace)

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 88, in execfile
    exec(compile(open(filename, 'rb').read(), filename, 'exec'), namespace)

  File "/home/ana/.spyder2-py3/temp.py", line 15, in <module>
    model.fit(values)

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/pgmpy-0.1.0-py3.4.egg/pgmpy/models/BayesianModel.py", line 569, in fit
    cpds_list = estimator.get_parameters()

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/pgmpy-0.1.0-py3.4.egg/pgmpy/estimators/MLE.py", line 65, in get_parameters
    state_counts.values[:, np.newaxis])

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/pgmpy-0.1.0-py3.4.egg/pgmpy/factors/CPD.py", line 139, in __init__
    super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))

  File "/home/ana/anaconda2/envs/pgmpy-env/lib/python3.4/site-packages/pgmpy-0.1.0-py3.4.egg/pgmpy/factors/Factor.py", line 100, in __init__
    raise TypeError("Values: Expected type int or type float, got ", values.dtype)
```

I found the same problem [ Page 97 example error #633 ](https://github.com/pgmpy/pgmpy/issues/633) and it seems to be fixed (the issue is closed), at least for windows users.

I am using debian (virtual machine on a windows host).

**Comment by khalibartan** [comment]
@anapinkg Are you using the latest version of pgmpy. If not please update the version, and report if the issues persists over the thread.

**Comment by anapinkg** [comment]
I installed the pgmpy this afternoon (following the instructions [here](http://pgmpy.org/install.html) so i guess i am using the latest version.

**Comment by khalibartan** [comment]
Okay. @yashu-seth Can you check this issue. I guess you use a windows platform?
@anapinkg We will try to look into things.We are working on removing pandas support so, this will be fixed anyways. And on my debian platform it ran just fine without error. 

 A simple fix I can guess, is convert the pandas data frame values to np.float . I guess it should run fine.

**Comment by yashu-seth** [comment]
@khalibartan @anapinkg I will look into this as soon as possible.

**Comment by anapinkg** [comment]
@khalibartan @yashu-seth  So, it is probably a plataform dependent issue...
In my case, the default int value is int64.

I did some debug and around line 79 of MLE.py there is this command:
`cpd = TabularCPD(node, var_card, np.array(values), evidence=parents, evidence_card=parent_card.astype('int'))`

In my execution, the problem starts with `np.array(values)` as this will assume the default int type.

---

I tried float with `np.random.uniform()`, and indeed, the mentioned error does not occur. Instead, i got 

> ValueError: Values array must be of size: 1000000

I believe that the MLE.get_parameters() is calculating wrong the `parent_card`  and `var_card`. But in this case, maybe i am misusing the package.

**Comment by eran-str** [comment]
I'm a windows user too, ran into the same problem.
I think that Pandas in window, the method values_count() is returning now a dtype.int64 and not  int,  
After changing the estimator\base.py base class to cast int in line 22:
self.node_card = {_node: int (get_node_card(_node, data)) for _node in self.model.nodes()}
and fixed another line of check error about dtype !=dtype.int64,
It seems to work properly for the above example.

I hope that helped :-)

**Comment by khalibartan** [comment]
@yashu-seth Can you look into this bug. It had been already reported more than once. I cannot debug it as I'm not able to reproduce the bug.

**Comment by yashu-seth** [comment]
I'll look into this today most probably, was busy with my end semesters.
But I'm free now.

**Comment by khalibartan** [comment]
@ankurankan Close this issue. Fixed in #682
